core/thread_announces.go

```diff
@@ -56,16 +56,14 @@ func (t *Thread) handleAnnounceBlock(hash mh.Multihash, block *pb.ThreadBlock) (
 	}
 
 	// update author info
-	pid, err := peer.IDB58Decode(block.Header.Author)
-	if err != nil {
-		return nil, err
-	}
-	inboxes := make([]repo.Cafe, 0)
-	for _, i := range msg.Inboxes {
-		inboxes = append(inboxes, protoCafeToModel(*i))
-	}
-	if err := t.addOrUpdatePeer(pid, block.Header.Address, msg.Username, inboxes); err != nil {
-		return nil, err
+	if msg.Contact != nil {
+		pid, err := peer.IDB58Decode(block.Header.Author)
+		if err != nil {
+			return nil, err
+		}
+		if err := t.addOrUpdatePeer(pid, protoContactToModel(msg.Contact)); err != nil {
```

Comment:
also pulled in from PubSub Contacts PR... Joins and accounces contain the _full_ contact object